src/Types.ts

```diff
@@ -160,65 +139,51 @@ export interface Document {
   collection_id?: string;
 
   meta_data?: {
-
     protocol: Protocol;
 
     url: string;
 
     meta?: {
-
       author?: string;
-
     };
-
   };
 
   collection_metadata?: {
-
     revision_number: number;
 
     document_id: string;
-
   };
-
 }
 
 export interface DocumentWithContext extends Document {
-
   context: DocumentContext;
-  
-  layers: DocumentLayer[];
 
+  layers: DocumentLayer[];
 }
 
 export interface DocumentContext {
+  id: string | undefined;
```

Comment:
If they are optional, you can also use `id?: string`. I believe those two notations are identical, as far as the effect on auto-completion goes. Just mentioning it because in this particular case we have both notations in the same interface.